modules/host/network.go

```diff
@@ -49,11 +49,11 @@ func (h *Host) initNetworking(address string) (err error) {
 		return err
 	}
 	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return err
+	}
 	go func() {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
```

Comment:
Are you sure that this causes a deadlock?

1. This thread is spawned, but put in the queue, `Add()` is not called yet.
2. Host closes, full clean shutdown occurs
3. This thread is reached. `Add()` returns an error, thread closes.

No deadlock, as far as I can tell. This goroutine will not shut down until after full clean shutdown, but there's no deadlock.